app/controllers/users/registrations_controller.rb

```diff
@@ -0,0 +1,14 @@
+class Users::RegistrationsController < Devise::RegistrationsController
+  respond_to :json
+
+  private
+
+  def respond_with(resource, _options = {})
+    if resource.persisted?
+      render json: { status: { code: 200, message: 'Signed up successfully.', data: resource } }, status: :ok
```

Comment:
Kindly consider changing the status code to `201`  & `:created` instead of `200` & `:ok` when creating a new user,
as per `http` conventions. Also it provides a clear sense of what the `request` entailed.